router_netconsole.c

```diff
@@ -40,6 +46,65 @@ struct netconsole_priv {
 	enum netconsole_state state;
 };
 
+static int ap121f_detect_main(void (*priv)__attribute__((unused)),
+			      const char *packet_buff, int packet_buff_len)
+{
+	struct ether_arp *arphdr;
+	int ret = 0;
+
+	if (!len_check(packet_buff_len, sizeof(struct ether_arp), "ARP"))
+		goto out;
+
+	arphdr = (struct ether_arp *)packet_buff;
+	if (arphdr->ea_hdr.ar_op != htons(ARPOP_REQUEST))
+		goto out;
+
+	if (*((unsigned int *)arphdr->arp_spa) != htonl(ap121f_ip))
+		goto out;
+
+	if (*((unsigned int *)arphdr->arp_tpa) != htonl(my_ip))
+		goto out;
+
+	if (memcmp(arphdr->arp_sha, ap121f_mac, ETH_ALEN))
+		goto out;
+
+	if (memcmp(arphdr->arp_tha, zero_mac, ETH_ALEN))
+		goto out;
+
+	ret = 1;
+
+out:
+	return ret;
+}
+
+static void netconsole_detect_post(struct node *node, const char *packet_buff,
+				   int packet_buff_len)
+{
+	struct netconsole_priv *priv = node->router_priv;
+	struct ether_arp *arphdr;
+
+	if (!len_check(packet_buff_len, sizeof(struct ether_arp), "ARP"))
+		goto out;
+
+	arphdr = (struct ether_arp *)packet_buff;
+
+	node->flash_mode = FLASH_MODE_NETCONSOLE;
+	node->his_ip_addr = load_ip_addr(arphdr->arp_spa);
+	node->our_ip_addr = load_ip_addr(arphdr->arp_tpa);
+	priv->state = NETCONSOLE_STATE_WAITING;
+
+out:
+	return;
+}
+
+const struct router_type ap121f = {
+	.desc = "Alpha AP121F",
```

Comment:
Told you before but you still didn't change the company name

Reply:
ops you are right. I am a hard nut to crack. I will fix that.